Testing: Add `needs_iam` marker to run tests with the IAM profile

0xquark opened the pull request:
<!-- Please read https://rucio.cern.ch/documentation/contributing before submitting a pull request -->
Fixes #7933 

Most of our test suites (both locally and in CI) use profiles, I introduced a `DEV_PROFILES` variable to get the profiles while running the tests and skip tests with the markers that doesn't have the profile required to run the test while also warning the user about it. 

Only these three tests need IAM profiles:
```
-  test_conveyer::test_transfer_with_tokens
-  test_reaper::test_deletion_with_tokens
-  test_Conveyer::test_bittorrent_submission
```

0xquark committed:
Testing: introduce 'needs_iam' marker and update integration workflow; ref #7933

- Added 'needs_iam' marker to tests requiring the IAM profile.
- Implemented logic to skip tests if the IAM profile is not available in the dev environment.
- Updated integration test workflow to export DEV_PROFILES environment variable for container setup.
- Modified docker-compose configuration to include DEV_PROFILES in the environment variables

rdimaio reviewed (commented):
To check the changes: we can only check `test_deletion_with_tokens`, because the other test is currently being skipped due to an existing ticket.

`test_deletion_with_tokens` is being run as expected here:
https://github.com/rucio/rucio/actions/runs/17066724292/job/48385560557?pr=7940#step:21:29

I guess we don't have any end to end setups where iam is not set up. Could you run this locally without the iam profile to see if the exclusion is working as expected? Or possibly you could run a gh workflow without spinning up the iam profile (so changing `workflows/integration_tests`, to check this

0xquark committed:
testing: add 'needs_iam' marker to test_bittorrent_submission for IAM dependency

While running the integration tests, The BitTorrent driver tries to get an OIDC token from indigoiam which creates an iam dependency.

0xquark commented:
> To check the changes: we can only check `test_deletion_with_tokens`, because the other test is currently being skipped due to an existing ticket.

For `test_transfer_with_tokens`, I removed the other markers and only included the`needs_iam` marker  to test when running it inside a dev container with the default profile. It worked as expected. 
<img width="1493" height="289" alt="image" src="https://github.com/user-attachments/assets/d68bc031-e89a-4290-b79d-7ca0181b1bc5" />

Similarly for `test_deletion_with_tokens` , I tested it locally using the dev-container which worked as expected.
<img width="1482" height="518" alt="image" src="https://github.com/user-attachments/assets/4e980fe3-7988-4c21-974c-1fca02d82d21" />

Also noticed that the `test_bittorrent_submission` also requires iam as it tries to get the OIDC tokens, which fails when the IAM profile is not running. Therefore, Added the marker for this test as well.

<img width="1322" height="330" alt="image" src="https://github.com/user-attachments/assets/03105296-968c-43eb-b194-faff2ca3e284" />

Apologies for not including the screenshots before.

> 
> `test_deletion_with_tokens` is being run as expected here: https://github.com/rucio/rucio/actions/runs/17066724292/job/48385560557?pr=7940#step:21:29
> 
> I guess we don't have any end to end setups where iam is not set up. Could you run this locally without the iam profile to see if the exclusion is working as expected? Or possibly you could run a gh workflow without spinning up the iam profile (so changing `workflows/integration_tests`, to check this
 
I've ran integration tests without the `iam` profile, lets see if there are any more places we need to make changes.

rdimaio commented:
> > To check the changes: we can only check `test_deletion_with_tokens`, because the other test is currently being skipped due to an existing ticket.
> 
> For `test_transfer_with_tokens`, I removed the other markers and only included the`needs_iam` marker to test when running it inside a dev container with the default profile. It worked as expected. <img alt="image" width="1493" height="289" src="https://private-user-images.githubusercontent.com/84588720/480547505-d68bc031-e89a-4290-b79d-7ca0181b1bc5.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTU4NTQ2MTMsIm5iZiI6MTc1NTg1NDMxMywicGF0aCI6Ii84NDU4ODcyMC80ODA1NDc1MDUtZDY4YmMwMzEtZTg5YS00MjkwLWI3OWQtN2NhMDE4MWIxYmM1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODIyVDA5MTgzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUyOWNkMTMxYzM1NTBjYTBhN2M2OWNhZWI3NzY2ZjRkZGE5MWFmOTlkM2RmZTJiOWY1ZTkxZTNhNGUxYjFhODkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.dv_C67EbTpLKqJM5YZb2UlbEHv5GFxu_aF2cL2Uhhi4">
> 
> Similarly for `test_deletion_with_tokens` , I tested it locally using the dev-container which worked as expected. <img alt="image" width="1482" height="518" src="https://private-user-images.githubusercontent.com/84588720/480548892-4e980fe3-7988-4c21-974c-1fca02d82d21.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTU4NTQ2MTMsIm5iZiI6MTc1NTg1NDMxMywicGF0aCI6Ii84NDU4ODcyMC80ODA1NDg4OTItNGU5ODBmZTMtNzk4OC00YzIxLTk3NGMtMWZjYTAyZDgyZDIxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODIyVDA5MTgzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRkM2EyNmQ0ZWZiNGI3ZjYyOGFmYWNjNzU4OTVhMTdhMjlmOTI1MThlMGE5YzIwZWUxNTM4MGExYjU3YjU2ZDQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.qm5vugjvUX3NwQmQFWZThZs9Xo8mc4RzyE-PgjKjoSU">
> 
> Also noticed that the `test_bittorrent_submission` also requires iam as it tries to get the OIDC tokens, which fails when the IAM profile is not running. Therefore, Added the marker for this test as well.
> <img alt="image" width="1322" height="330" src="https://private-user-images.githubusercontent.com/84588720/480580535-03105296-968c-43eb-b194-faff2ca3e284.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTU4NTQ2MTMsIm5iZiI6MTc1NTg1NDMxMywicGF0aCI6Ii84NDU4ODcyMC80ODA1ODA1MzUtMDMxMDUyOTYtOTY4Yy00M2ViLWIxOTQtZmFmZjJjYTNlMjg0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODIyVDA5MTgzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVhMzIwYWEyYWI3NDhhOWJjODhjNjgwNzhhNDZjN2RjN2VjNDAyNTA2Y2Q5OGQ5YTliOGQ0Y2NkZjJlN2Y4Y2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.p_-BiilTV8GUs8mNmB4bkILSR6vjPvr9HBh1nQKT5MA">
> 
> Apologies for not including the screenshots before.
> 
> > `test_deletion_with_tokens` is being run as expected here: https://github.com/rucio/rucio/actions/runs/17066724292/job/48385560557?pr=7940#step:21:29
> > I guess we don't have any end to end setups where iam is not set up. Could you run this locally without the iam profile to see if the exclusion is working as expected? Or possibly you could run a gh workflow without spinning up the iam profile (so changing `workflows/integration_tests`, to check this
> 
> I've ran integration tests without the `iam` profile, lets see if there are any more places we need to make changes.

Nice, thanks!
